homeassistant/components/rest/data.py

```diff
@@ -57,6 +59,9 @@ async def async_update(self, log_errors: bool = True) -> None:
 
         rendered_headers = template.render_complex(self._headers, parse_result=False)
         rendered_params = template.render_complex(self._params)
+        self._async_client._default_encoding = (  # pylint: disable=protected-access
+            self._encoding
+        )
```

Comment:
This is a very bad idea - it patches the "shared" instance of the HTTPX client.
This should be adjusted instead of line 56:
```python
            self._async_client = create_async_httpx_client(
                self._hass, verify_ssl=self._verify_ssl, default_encoding=XXX
            )
```

Reply:
Ok. Thanks a lot for your input. Tried to get it working with `get_async_client` but never found the way to do it. I changed to `create_async_httpx_client`.

I'm really new to all of this and my knowledge is very limited. Everything is working in my development environment, but I'm afraid to break things. Please double check this change.